aten/src/ATen/native/transformers/cuda/mem_eff_attention/kernels/generate_kernels.py

```diff
@@ -0,0 +1,403 @@
+                preload_mmas = is_half and sm >= 80 and output_in_rf
+                bj = 128 if (preload_mmas and max_k > 64) else 64
+                kernels.append(
+                    cls(
+                        aligned=aligned,
+                        dtype=dtype,
+                        sm_range=(sm, sm_max),
+                        apply_dropout=apply_dropout,
+                        preload_mmas=preload_mmas,
+                        block_i=bi,
+                        block_j=bj,
+                        max_k=max_k,
+                    )
+                )
+                # A few specialized kernels that are faster
+                if apply_dropout or max_k > 128 or not is_half or not aligned:
+                    continue
+                if sm not in [70, 80]:
+                    continue
+                kernels.append(
+                    cls(
+                        aligned=aligned,
+                        dtype=dtype,
+                        sm_range=(sm, sm_max),
+                        apply_dropout=apply_dropout,
+                        preload_mmas=preload_mmas,
+                        block_i=bi,
+                        block_j=bj,
+                        max_k=max_k,
+                        keys_queries_aligned_to_blocksizes=True,
+                    )
+                )
+        # Add some specialized kernels for stable diffusion BW (K=80)
+        # This is the only kernel that can keep the outputs on RF on
+        # Sm86/Sm89, so it's much faster than the 64x64 one
+        for dtype in ["f16", "bf16"]:
+            kernels.append(
+                cls(
+                    aligned=True,
+                    dtype=dtype,
+                    sm_range=(80, 90),
+                    apply_dropout=False,
+                    preload_mmas=True,
+                    block_i=128,
+                    block_j=64,
+                    max_k=96,
+                    # Sm80 has a faster kernel for this case
+                    dispatch_cond="cc == 86 || cc == 89",
+                )
+            )
+        return kernels
+
+
+T = TypeVar("T", FwdKernel, BwdKernel)
+
+# impl_file takes absolute paths
+
+
+def write_decl_impl(
+    kernels: List[T], family_name: str, impl_file: str, autogen_dir: Path, disable_def: str = None
+) -> None:
+    cpp_file_header = """/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+// This file is auto-generated. See "generate_kernels.py"
+"""
+
+    kernels.sort()
+
+    implfile_to_kernels: Dict[str, List[T]] = collections.defaultdict(list)
+    cat_to_kernels: Dict[Tuple[str, int, int], List[T]] = collections.defaultdict(list)
+
+    dispatch_all = ""
+    declarations = cpp_file_header + "#pragma once\n"
+    # declarations += f"#ifndef {disable_def}\n"
```

Comment:
We gate the building of these files in cmake, I don't think we need these global macros